worldengine/astar.py

```diff
@@ -158,11 +158,10 @@ def __init__(self, map_data, width, height):
     def get_node(self, location):
         x = location.x
         y = location.y
-        if x < 0 or x >= self.w or y < 0 or y >= self.h:
-            return None
-        d = self.m[(y * self.w) + x]
-
-        return Node(location, d, ((y * self.w) + x))
+        if (0 <= x < self.w) and (0 <= y < self.h):
```

Comment:
For your edification and ours, can you profile this after 100,000 iterations?

The original takes advantage of 'or' short-cutting so it doesn't have to evaluate the rest. The one on the right has to evaluate the whole thing.

I also believe that your parenthesis are redundant.